### ISSUE TITLE
Minor water area names are too heavily marked

### ISSUE THREAD

**Issue by michaelblyons** [ZeLonewolf/openstreetmap-americana#620]
Ponds and (oddly) swimming pools are given equal significance to substantial water bodies (and at any rate, more than they deserve).

See [this arbitrary spot in New Jersey](https://zelonewolf.github.io/openstreetmap-americana/#map=14/40.84167/-74.29898) for a pond. And if the pond wasn't there, I'm sure the [school swimming pool right next to it](https://zelonewolf.github.io/openstreetmap-americana/#map=17/40.841672/-74.298983) would have shown up instead.

**Comment by 1ec5** [comment]
For swimming pools specifically, we can tweak the font size for when `class` is `swimming_pool`:

https://github.com/ZeLonewolf/openstreetmap-americana/blob/478e36466b2f0ec0b5064b79bb0cf0ab9dbe9c2e/src/layer/water.js#L219-L224

To solve the issue of too-prominent waterbody labels generally, we’d need the vector tiles to indicate the relative size of these polygons. (It isn’t possible to compute the area client-side, because a single feature may have already been cut up along tile boundaries.)

**Comment by ZeLonewolf** [comment]
Beyond fixing the swimming pool problem, we could ask OpenMapTiles to encode `class=pond` for waterbodies tagged `water=pond`.  Currently, all lake/pond/reservoir are tagged `class=lake`.  This would give a clear differentiator between more important and less important waterbodies.

**Comment by 1ec5** [comment]
That could give us some additional flexibility, but the various `water` values don’t say a whole lot about size, which is the root of the issue here. A lot of rules in Mapnik/CartoCSS styles depend on a feature property that says roughly the area of the feature. Nothing like that exists in the vector tiles we’re using.

**Comment by zekefarwell** [comment]
There is much work to be done classifying different sizes of water features in OpenMapTiles.  All kinds of water features from tiny fountains and ponds to massive bays and reservoirs are all just `class=lake` in OMT. https://github.com/openmaptiles/openmaptiles/pull/1385 did split out swimming pools.  If OMT could calculate a rough area property as @1ec5 suggests, that would probably be more useful than classification.

**Comment by ZeLonewolf** [comment]
From recollection, encoding the area of a waterbody is not actually as hard as it might sound.  Are we able to do some sort of "if this zoom then render if < this size" logic if the water label POIs include that data?  Or should the tiles just suppress labels if the waterbody is too small at a particular zoom?  I think either approach is probably technically possible.

**Comment by adamfranco** [comment]
Having an area value for most area features would be really neat. This would theoretically allow showing labels to huge landuse/park/water/natural/etc without cluttering with labels for small ones.

I would imagine that each feature type would have a different threshold for area and zoom at which point it's labels become significant, with the current zoom thresholds for labels being the max zoom we're all are shown.

**Comment by zekefarwell** [comment]
Ideally any polygon in the tiles would have a numeric property indicating the rough area.  Then this could be used to make runtime styling decisions.  The decision to render a particular polygon would probably happen at a lower zoom level than the decision to render a label for that polygon.

**Comment by ZeLonewolf** [comment]
I believe PostGIS has function that would allow us to encode either real-world size or pixel size.

**Comment by michaelblyons** [comment]
Unblocked?

**Comment by ZeLonewolf** [comment]
Possibly.  This can at least be tested against some of the more recent vector servers listed at:
https://6ug7hetxl9.execute-api.us-east-2.amazonaws.com/

**Comment by ZeLonewolf** [comment]
The servers "planet-low-zoom" and "uk-ireland" can be used for testing low- and high-zoom functionality, respectively.

**Comment by michaelblyons** [comment]
Did the underlying data ever change to support this and/or #530?

**Comment by wmisener** [comment]
I don't think so. As far as I can tell, in the OMT `water` layer, ponds and swimming pools are now their own classes. But I think we only render the biggest water bodies from the `water` layer, that is, those water bodies that actually get labeled along their shape. For water bodies small enough to be rendered as points, we populate their names from the `water_name` layer, which still classifies all ponds and swimming pools as `lakes`. I also don't see any area information in the tiles, so I don't think there's currently a way to distinguish these in the `water_name` layer. See the small swimming pool in the screenshots as an example.
<img width="393" alt="Screenshot 2024-11-20 at 11 10 21 PM" src="https://github.com/user-attachments/assets/b3d1188a-0246-4bc0-8e04-07723255940f">
<img width="179" alt="Screenshot 2024-11-20 at 11 09 51 PM" src="https://github.com/user-attachments/assets/922d95b3-fb39-4e03-8dff-d9abfed5622b">
